torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -651,20 +651,22 @@ std::unordered_map<
 std::unordered_set<Symbol> IrParser::jit_reduction_op_registry_;
 bool IrParser::init_registry_ = true;
 
-ProfileIValueOp* insertProfileIValueOp(Node* node, size_t offset) {
-  auto pn = new ProfileIValueOp(node->owningGraph(), nullptr);
+ProfileIValueOp* insertProfileIValueOp(
+    Node* node,
+    size_t offset,
+    ProfilingRecord* pr) {
   auto in_val = node->input(offset);
+  auto pn = pr->createProfileIValueNode(in_val);
   pn->insertBefore(node);
-  pn->addInput(in_val);
-  auto pno = pn->addOutput();
-  pno->setType(in_val->type());
-  in_val->replaceAllUsesAfterNodeWith(pn, pno);
-  pn->ty_(attr::profiled_type, in_val->type());
+  // TODO: I think this is better
+  node->replaceInput(offset, pn->output());
+  // TODO: why is this here?
+  // pn->ty_(attr::profiled_type, in_val->type());
```

Comment:
I thought we need to know the profiled type as well.
I don't how to retrieved it from `attr::ivalue`.

Reply:
I'd just leave it up to `ProfilingRecord::instrumentBlock` to insert `insertShapeProfileOp`. `prim::profile` is weird and different enough from either `specializeAutogradZero` or `parser.cpp`.

Reply:
I was referring to type of scalar value. This is not trying to replace profiling of TensorType.
I was basically using it to determine what scalar to cast the ivalue to.
https://github.com/csarofeen/pytorch/blob/sum_to_size_PR/torch/csrc/jit/codegen/cuda/graph_fuser.cpp#L46-L58

Reply:
could we consider giving them different names `attr::profiled_axes`, `attr::cond` ? It looks like we are trying to tell apart axes and a isCuda/isTraining boolean? or are we have an attribute that changes its type?

Reply:
i can make a change if you think this could work

Reply:
that should work.
We might need to add another one later (`attr::profiled_reduction_shapes`) for `_grad_sum_to_size`. Let's not worry about it for now.